tests/keras/backend/backend_test.py

```diff
@@ -1702,6 +1702,56 @@ def test_sparse_concat(self):
             assert k_s_d.shape == k_d.shape
             assert_allclose(k_s_d, k_d, atol=1e-05)
 
+    @pytest.mark.skipif((K.backend() != 'mxnet'),
+                        reason='Testing only for MXNet backend')
```

Comment:
There is nothing specific for mxnet, we can test tensorflow backend as well.

Reply:
the eval() implementation is different for mxnet - eval() converts the array into a numpy array - making it dense & fails the test. In Tensorflow the original class of the sparse tensor doesn't change - https://github.com/tensorflow/tensorflow/blob/r1.10/tensorflow/python/ops/sparse_ops.py#L724

Reply:
It's the same for both backends, eval() are supposed to convert to dense and return a numpy array. same in tensorflow backend.  See [tf_backend](https://github.com/awslabs/keras-apache-mxnet/blob/master/keras/backend/tensorflow_backend.py#L672)

Reply:
With MXNet backend, when we try to evaluate the value of a sparse KerasSymbol - the dense value is returned and because of this `is_sparse(eval(KerasSymbol))` is always `False` - we basically [return the numpy array](https://github.com/awslabs/keras-apache-mxnet/blob/master/keras/backend/mxnet_backend.py#L565) which is always dense.
In TF backend, a `SparseTensor` class has been used and because of this even when the tensor is converted to dense it always remains [an instance of the SparseTensor class](https://github.com/awslabs/keras-apache-mxnet/blob/master/keras/backend/tensorflow_backend.py#L330)

Reply:
Will this behavior not become an issue during training when weights are sparse tensor?

Reply:
I dont think this should be an issue. We are just abstracting the _forward_pass() operation in eval() and then converting it to a numpy array.